flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPool.java

```diff
@@ -1518,11 +1469,27 @@ void clear() {
 				Collection<TaskManagerLocation> preferredLocations) {
 
 			final SlotRequestId requestId = new SlotRequestId();
+			Collection<AllocationID> previousAllocationIDs = Collections.emptyList();
+
+			// try to extract previous allocation ids, if applicable, so that we can reschedule to the same slot
+			Execution execution = task.getTaskToExecute();
```

Comment:
Don't let us couple the Execution with the `SlotPool`. Let's give the `SlotProfile` to this method instead.

Reply:
👍